collector-api/internal/api/snapshots.go

```diff
@@ -190,10 +208,126 @@ func CompactSnapshotHandler(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprint(w, "Compact snapshot successfully processed")
 }
 
-// Dummy function for handling the compact snapshot
-func handleCompactSnapshot(s3Location string, collectedAt int64) error {
-	// Here, implement your actual logic for processing the compact snapshot
-	// e.g., saving to local storage, uploading to S3, etc.
-	log.Printf("Processing compact snapshot with s3_location: %s and collected_at: %d", s3Location, collectedAt)
+// extractSystemInfo extracts system information from request headers
+func extractSystemInfo(r *http.Request) map[string]string {
+	systemInfo := make(map[string]string)
+	headers := []string{
+		"Pganalyze-System-Id",
+		"Pganalyze-System-Id-Fallback",
+		"Pganalyze-System-Scope",
+		"Pganalyze-System-Scope-Fallback",
+		"Pganalyze-System-Type",
+		"Pganalyze-System-Type-Fallback",
+	}
+
+	for _, header := range headers {
+		value := r.Header.Get(header)
+		if value != "" {
+			key := strings.ToLower(strings.TrimPrefix(header, "Pganalyze-"))
+			key = strings.ReplaceAll(key, "-", "_")
+			systemInfo[key] = value
+		}
+	}
+
+	return systemInfo
+}
+
+var previousBackends map[BackendKey]bool
+
+// handleCompactSnapshot processes a compact snapshot, generates metrics and stale markers,
+// and sends them to Prometheus
+func handleCompactSnapshot(cfg *config.Config, s3Location string, collectedAt int64, systemInfo map[string]string) error {
+	if cfg.Debug {
+		log.Printf("Processing compact snapshot with s3_location: %s and collected_at: %d", s3Location, collectedAt)
+	}
+
+	f, err := os.Open(path.Join("/usr/local/autodba/share/collector_api_server/storage", s3Location))
```

Comment:
should this path be pulled from config ?

Reply:
Thanks for pointing this out. Fixed in 60c73f884f93acde1b9adbe309b1f73c2d7348dc